src/main/java/duke/command/Parser.java

```diff
@@ -0,0 +1,188 @@
+            case ("delete"):
+            case("find"):
+                return this.processList(com, index);
+
+            case ("todo"):
+            case ("deadline"):
+            case ("event"):
+                return this.processTask(com, task, date);
+            }
+        }
+        throw new DukeException("Sorry, I did not understand: " + command
+                + ".\nUse \"help\" to look at available commands.");
+    }
+
+    /**
+     * Helper function to process Task related commands.
+     * @param com The type of Task.
+     * @param task The description of the Task.
+     * @param date The date of the Task to be done at or by (if applicable).
+     * @return String result of the function executed.
+     * @throws DukeException A custom Exception that carries a message for the user if thrown.
+     */
+    private String processTask(String com, String task, String date) throws DukeException {
+
+        switch(com) {
+        case("todo"):
+        if (!task.equals("")) {
+                return list.addItem(new Todo(task));
+            } else {
+                throw new DukeException("Please write a task to be done, with \"todo <task>\"");
+            }
+        case("deadline"):
+            if (!task.equals("")) {
+                try {
+                    return list.addItem(new Deadline(task, Parser.convertDate(date)));
+                } catch (DateTimeParseException e){
+                    throw new DukeException("Please write your date in the format \"dd/MM/yyyy\"");
+                }
+            } else {
+                throw new DukeException("Please write a deadline, with \"deadline <task> /by <date>\"");
+            }
+        case("event"):
+            if (!task.equals("")) {
+                try {
+                    return list.addItem(new Event(task, Parser.convertDate(date)));
+                } catch (DateTimeParseException e) {
+                    throw new DukeException("Please write your date in the format \"dd/MM/yyyy\"");
+                }
+            } else {
+                throw new DukeException("Please write an event, with \"event <task> /at <date>\"");
+            }
+        default:
+            return null;
+        }
+    }
+
+    /**
+     * Helper function to process list related commands.
+     * @param com The command keyword that interacts with the list.
+     * @param index The index of which the command wishes to act on.
+     * @return String result of the function executed.
+     * @throws DukeException A custom Exception that carries a message for the user if thrown.
+     */
+    private String processList(String com, String index) throws DukeException {
+
+        switch(com) {
+        case("done"):
+            if (!index.equals("")) {
+                return list.markDone(Integer.parseInt(index) - 1);
+            } else {
+                throw new DukeException("Please choose a task to mark as done, with \"done <task number>\"");
+            }
```

Comment:
Will it be better to include a keyword break here? I have also noticed some similar problems in several other places.